app/controllers/courses_controller.rb

```diff
@@ -29,7 +29,13 @@ def index
     elsif params[:tab] == 'featured'
       @courses = @courses.where(featured: true)
     elsif params[:copy_courses]
-      @courses = @courses.reorder(featured: :desc, year: :desc, name: :asc)
+      @featured_courses = @courses.where(featured: true)
+      @own_courses = @courses.where(featured: false)
+                             .reorder(year: :desc, name: :asc)
+                             .select { |course| current_user.course_admin?(course) }
+      @other_courses = @courses.where(featured: false)
+                               .reorder(year: :desc, name: :asc)
+                               .reject { |course| current_user.course_admin?(course) }
```

Comment:
```suggestion
      featured_courses = @courses.where(featured: true)
      own_courses = @courses.where(featured: false)
                             .reorder(year: :desc, name: :asc)
                             .select { |course| current_user.course_admin?(course) }
      other_courses = @courses.where(featured: false)
                               .reorder(year: :desc, name: :asc)
                               .reject { |course| current_user.course_admin?(course) }
      @courses = featured_courses + own_courses + other_courses
```
This way, no changes to the HTML are necessary. (Make sure to check this locally, I'm not entirely sure if the `+` works if the actual data hasn't been loaded yet.)

Reply:
This seems to result in an error: 
![Screenshot from 2021-08-30 16-50-11](https://user-images.githubusercontent.com/53220653/131358614-c20f583b-b2bd-488e-9387-43094a990c38.png)

It would be nice if that was possible because now there is a lot of redundant code in the HTML

Reply:
You can look at the server log to see the actual error. Probably an explicit load is necessary for the `featured_courses`.